openlawteam/tribute-discord-service
# ENG-1877: Investment Poll Bot

[ENG-1877](https://tributelabs.atlassian.net/browse/ENG-1875)

In brief, the opportunity poll bot allows operators to setup a new poll, using the `/investment` command prompting two numerical inputs minimum and maximum allocation amounts. Afterward, a select menu will be provided containing a list of active opportunities for the DAO. Upon, selecting an opportunity, the poll will render buttons for the allocation range. 

The test bot in Tribute Labs Team discord server uses data for test v2 dao.  

See PRD and prototype video for further details.

🥳 **Adds**

- `investmentHandler` for new `/investment` command
- Investment services for `http-api` endpoints
- Unit tests for services and investment command
- `MONGO_DB_URI` env var and config
- MongoDb docker test setup added GH workflow
- Koa Cors module added to http server

**Test Plan**

1. Add `BOT_TOKEN_TRIBUTE_TOOLS` and `MONGO_DB_URI` to .env . Contact Sophia for the `BOT_TOKEN_TRIBUTE_TOOLS` token. 
2. Setup up `daosLocalhost.example.ts` as per the README.md
3. `npm ci && npm start`
4. Go to discord channel `discord-service-test` https://discord.com/channels/722525233755717762/886976610018934824
5. Enter command `/investment` under allocation amounts. Select an opportunity.
6. Click on a button and see the vote recorded result in the terminal. 

[ENG-1877]: https://tributelabs.atlassian.net/browse/ENG-1877?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
